stock_location_archive_constraint/models/product_product.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.constrains("active")
+    def _check_active(self):
+        res = self.env['stock.quant'].search(
+            [
+                ('product_id', 'in', self.ids),
+                ('quantity', '!=', 0.0)
+            ], limit=1
+        )
+        if res:
+            raise ValidationError(
+                _("It is not possible to archive a product that has quant associated.")
+            )
+        res = self.env['stock.move'].search(
+            [
+                ('product_id', 'in', self.ids),
+                ('state', 'not in', ('done', 'cancel'))
+            ], limit=1
+        )
+        if res:
+            raise ValidationError(
+                _("It is not possible to archive a location that has "
```

Comment:
This is not archiving a location, but a product...

And thus, the module name should change, because this is checking something more. Maybe `stock_archive_constraint`.